pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -150,7 +158,24 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
 		}
 		return reconcile.Result{}, err
 	}
+	deletetimeStamp := engine.Instance.ObjectMeta.GetDeletionTimestamp()
 
+	if engine.Instance.Spec.EngineStatus == "stop" ||
+		engine.Instance.Spec.EngineStatus == "stopped" ||
+		deletetimeStamp != nil {
+		return r.reconcileForDelete(request)
+	} else if engine.Instance.Spec.EngineStatus == "start" {
+		r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "Started reconcile for chaosEngine", "Will start Chaos Runner & Chaos Monitor")
+		err := r.addFinalzerToEngine(engine, "chaosengine.litmuschaos.io/finalizer")
```

Comment:
```suggestion
		err := r.addFinalzer(engine, "chaosengine.litmuschaos.io/finalizer")
```

Reply:
Its seems a little generic, let the developer know from function name itself, which resource is getting the finalizer

Reply:
okay